caas/kubernetes/provider/k8swatcher.go

```diff
@@ -47,32 +48,65 @@ func newKubernetesNotifyWatcher(wi watch.Interface, name string, clock jujuclock
 const sendDelay = 1 * time.Second
 
 func (w *kubernetesNotifyWatcher) loop() error {
+	signals := make(chan struct{})
+	defer close(signals)
```

Comment:
Leave the channel open. A send could occur after this method has returned, because we don't explicitly wait for the informer's go-routine to exit.